### PR TITLE
AzureMonitor: Add Container Insights Syslog Dashboard

### PR DESCRIPTION
**Pull request by JohnJMartins**
**What is this feature?**

Azure Monitor - Container Insights offers the ability to collect Syslog events from Linux nodes in your Azure Kubernetes Service (AKS) clusters. This includes the ability to collect logs from control plane components like kubelet. Customers can also use Syslog for monitoring security and health events, typically by ingesting syslog into a SIEM system like Microsoft Sentinel.

SS1

**Why do we need this feature?**

With this Syslog Grafana dashboard from Azure Monitor, users can get a quick snapshot of syslog messages. At a glance, users can see if there's a large number of critical or warning messages. The dashboards also breaks down syslog message counts by host, process, and facility name - making it easier understand where errors might be coming from. Additionally, users can see the 50 most recent Syslog messages and click through to access the full Syslog table in Log Analytics. 

**Who is this feature for?**

Anyone looking for monitoring of container insights syslog metrics

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.

### CODE REVIEW

File: public/app/plugins/datasource/azuremonitor/dashboards/containerInsightsSyslog.json [hunk 1]
```diff
@@ -0,0 +1,1291 @@
...
+          "y": 3
+        },
+        "id": 18,
+        "options": {
+          "code": {
+            "language": "plaintext",
+            "showLineNumbers": false,
+            "showMiniMap": false
+          },
+          "content": "This dashboard visualizes Syslog collected by Azure Monitor - Container Insights. Read more in the Syslog documentation here:   [aka.ms/CISyslog](https://aka.ms/CISyslog). \n\nIn case of issues, please ensure that \n1. The [Azure Monitor data source is enabled](https://learn.microsoft.com/azure/managed-grafana/how-to-data-source-plugins-managed-identity?tabs=azure-portal).\n2. [Syslog is enabled](https://aka.ms/cisyslog#how-to-enable-syslog) in Container Insights  \n3. You have chosen an *LAWorkspace* and *ClusterName* above which have Syslog data flowing. *Subscription* is selected automatically. \n4. The time range is correct \n\nRead more here: [aka.ms/CISyslogGrafana](https://aka.ms/CISyslogGrafana)",
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/dashboards/containerInsightsSyslog.json:89]:
```suggestion
          "content": "This dashboard visualizes Syslog data collected by Azure Monitor - Container Insights. Read more in the Syslog documentation here:   [aka.ms/CISyslog](https://aka.ms/CISyslog). \n\nIn case of issues, please ensure that \n1. An [Azure Monitor data source is configured](https://grafana.com/docs/grafana/latest/datasources/azure-monitor/#configure-the-data-source).\n2. [Syslog is enabled](https://aka.ms/cisyslog#how-to-enable-syslog) in Container Insights  \n3. You have chosen an *LAWorkspace* and *ClusterName* above which have Syslog data flowing. *Subscription* is selected automatically. \n4. The time range is correct \n\nRead more here: [aka.ms/CISyslogGrafana](https://aka.ms/CISyslogGrafana)",
```

File: public/app/plugins/datasource/azuremonitor/dashboards/containerInsightsSyslog.json [hunk 2]
```diff
@@ -0,0 +1,1291 @@
+{
+    "__inputs": [],
+    "__elements": {},
+    "__requires": [
+      {
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/dashboards/containerInsightsSyslog.json:5]: This panel won't be available in Grafana generally and so should be removed.

**Review by aangelisc — CHANGES_REQUESTED** [review]
Thanks for the contribution @JohnJMartins. I've requested some changes to make this dashboard more generalised. You'll also need to run `prettier` on the dashboard file before this can be merged otherwise it'll fail CI 😊

File: public/app/plugins/datasource/azuremonitor/dashboards/containerInsightsSyslog.json (hunk 1 above)
Comment (JohnJMartins) [reply public/app/plugins/datasource/azuremonitor/dashboards/containerInsightsSyslog.json:89]: updated this to be more general

File: public/app/plugins/datasource/azuremonitor/dashboards/containerInsightsSyslog.json (hunk 2 above)
Comment (JohnJMartins) [reply public/app/plugins/datasource/azuremonitor/dashboards/containerInsightsSyslog.json:5]: I am a bit confused here on which lines I should be removing. Also, I have run prettier on the file on my end and am not seeing any updates to the formatting, is there something else I should be doing?

Comment (aangelisc) [reply public/app/plugins/datasource/azuremonitor/dashboards/containerInsightsSyslog.json:5]: R.E. prettier are you using the command available in Grafana? `yarn prettier:write`

Comment (JohnJMartins) [reply public/app/plugins/datasource/azuremonitor/dashboards/containerInsightsSyslog.json]: Believe I have corrected the formatting issues. Please let me know if there's anything else you need me to do before we can get this merged!

**Review by aangelisc — APPROVED** [review]
This all LGTM now! Thanks @JohnJMartins!

**Comment by aangelisc** [comment]
Hey @JohnJMartins, would you mind merging main into this branch please?

**Comment by JohnJMartins** [comment]
@aangelisc Should be merged now! Let me know if that works.

**Comment by aangelisc** [comment]
All good @JohnJMartins, thanks again 😊

**Pull request merged**